Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -196,27 +200,44 @@ public struct Binary: BSONValue, Equatable, Codable, Hashable {
     internal var bson: BSON { .binary(self) }
 
     /// The binary data.
-    public let data: Data
+    public let data: ByteBuffer
 
     /// The binary subtype for this data.
-    public let subtype: UInt8
+    public let subtype: Subtype
 
     /// Subtypes for BSON Binary values.
-    public enum Subtype: UInt8 {
+    public struct Subtype: Equatable, Codable, Hashable, RawRepresentable {
         /// Generic binary subtype
-        case generic,
-            /// A function
-            function,
-            /// Binary (old)
-            binaryDeprecated,
-            /// UUID (old)
-            uuidDeprecated,
-            /// UUID (RFC 4122)
-            uuid,
-            /// MD5
-            md5,
-            /// User defined
-            userDefined = 0x80
+        public static let generic = Subtype(0x00)
+        /// A function
+        public static let function = Subtype(0x01)
+        /// Binary (old)
+        public static let binaryDeprecated = Subtype(0x02)
+        /// UUID (old)
+        public static let uuidDeprecated = Subtype(0x03)
+        /// UUID (RFC 4122)
+        public static let uuid = Subtype(0x04)
+        /// MD5
+        public static let md5 = Subtype(0x05)
+        /// Encrypted BSON value
+        public static let encryptedValue = Subtype(0x06)
+
+        /// Subtype indicator value
+        public var rawValue: UInt8
+
+        public init(_ value: UInt8) { self.rawValue = value }
+        public init?(rawValue: UInt8) { self.rawValue = rawValue }
```

Comment:
sorry for the back and forth but I think maybe we should omit `RawRepresentable` conformance because the public initializers here allows users to bypass the validation of their custom subtype value. This is not great because a user could start using eg 0x07 as a custom subtype and then we could add a conflicting subtype in a future server version. 0x07-0x7F are reserved for future non-custom subtypes.

Reply:
Yea I think it would make more sense to not conform, I'll revert

Reply:
Sorry to reignite this conversation, but could we not just return `nil` if a user specified a reserved but unused value? From the docs of `RawRepresentable`, this seems like the ideal case to conform to it (i.e. modeling a C style enum).

Reply:
that is a good point, didn't think of that lol. i'm fine with returning nil for reserved subtypes